samples-lib/src/main/java/com/esri/arcgismaps/sample/sampleslib/DownloaderActivity.kt

```diff
@@ -224,59 +231,56 @@ abstract class DownloaderActivity : AppCompatActivity() {
+            ) { totalBytes, bytesRead ->
+                if (totalBytes != null) {
+                    // update the download percentage progress
+                    val percentage = ((100.0 * bytesRead) / totalBytes).roundToInt()
+                    lifecycleScope.launch(Dispatchers.Main) {
+                        progressIndicator.progress = percentage
+                        progressTV.text = "$percentage%"
+                    }
                 }
-                // set up the file at the download path
-                val destinationFilePath = provisionLocation.path + File.separator + portalItem.name
-                val provisionFile = File(destinationFilePath)
-                // create file at location to write the PortalItem ByteArray
-                provisionFile.createNewFile()
-                // create and write the file output stream
-                val writeOutputStream = FileOutputStream(provisionFile)
-                writeOutputStream.write(byteArray)
-
+            }.onSuccess {
                 // unzip the file if it is a .zip
                 if (portalItem.name.contains(".zip")) {
-                    val fileInputStream = FileInputStream(destinationFilePath)
-                    val zipInputStream = ZipInputStream(BufferedInputStream(fileInputStream))
-                    var zipEntry: ZipEntry? = zipInputStream.nextEntry
-                    val buffer = ByteArray(1024)
-                    while (zipEntry != null) {
-                        if(zipEntry.isDirectory){
-                            File(provisionLocation.path, zipEntry.name).mkdirs()
-                        }
-                        else {
-                            val file = File(provisionLocation.path, zipEntry.name)
-                            val fout = FileOutputStream(file)
-                            var count = zipInputStream.read(buffer)
-                            while (count != -1) {
-                                fout.write(buffer, 0, count)
-                                count = zipInputStream.read(buffer)
+                    // set up the file at the download path
+                    val provisionFile = File(destinationFilePath)
+                    // set up the input streams
+                    FileInputStream(destinationFilePath).use { fileInputStream ->
+                        ZipInputStream(BufferedInputStream(fileInputStream)).use { zipInputStream ->
+                            var zipEntry: ZipEntry? = zipInputStream.nextEntry
+                            val buffer = ByteArray(1024)
+                            while (zipEntry != null) {
+                                if (zipEntry.isDirectory) {
+                                    File(provisionLocation.path, zipEntry.name).mkdirs()
+                                } else {
+                                    val file = File(provisionLocation.path, zipEntry.name)
+                                    val fout = FileOutputStream(file)
```

Comment:
Should also make use of `use` here:
```suggestion
                                    FileOutputStream(file).use { fileOutputStream ->
```

Reply:
Yeah, it should! Also removed the `fout.close()` since `.use` should be a close the output stream.